registration-processor/pre-processor/registration-processor-packet-classifier-stage/src/main/java/io/mosip/registration/processor/stages/packetclassifier/utility/PacketClassifierUtility.java

```diff
@@ -0,0 +1,60 @@
+package io.mosip.registration.processor.stages.packetclassifier.utility;
+
+import java.util.List;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import io.mosip.kernel.core.exception.BaseCheckedException;
+import io.mosip.registration.processor.core.exception.util.PlatformErrorMessages;
+
+@Component
+public class PacketClassifierUtility {
+
+	/**
+	 * The mandatory languages that should be used when dealing with field type that
+	 * has values in multiple languages
+	 */
+	@Value("#{'${mosip.mandatory-languages}'.split(',')}")
+	private List<String> mandatoryLanguages;
+
+	/**
+	 * The optional languages that should be used when dealing with field type that
+	 * has values in multiple languages
+	 */
+	@Value("#{'${mosip.optional-languages}'.split(',')}")
+	private List<String> optionalLanguages;
+
+	/** The constant for language label in JSON parsing */
+	private static final String LANGUAGE_LABEL = "language";
+
+	public String getTagLanguageBasedOnFieldDTO(String value) throws JSONException, BaseCheckedException {
```

Comment:
Can we change this method to acually return the value based on language settings, rather than giving the language to avoid iterating the array multiple times ?

Rename this method to getLanguageBasedValueForSimpleType, return the value directly and call this method inside the simpleType swich case

Reply:
done changes as mentioned above...